packages/block-library/src/columns/edit.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { compose } from '@wordpress/compose';
+import {
+	PanelBody,
+	RangeControl,
+} from '@wordpress/components';
+import { Fragment } from '@wordpress/element';
+import {
+	InspectorControls,
+	InnerBlocks,
+	BlockControls,
+	BlockVerticalAlignmentToolbar,
+
+} from '@wordpress/editor';
+import { withSelect, withDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { getColumnsTemplate } from './utils';
+
+/**
+ * Allowed blocks constant is passed to InnerBlocks precisely as specified here.
+ * The contents of the array should never change.
+ * The array should contain the name of each block that is allowed.
+ * In columns block, the only block we allow is 'core/column'.
+ *
+ * @constant
+ * @type {string[]}
+*/
+const ALLOWED_BLOCKS = [ 'core/column' ];
+
+export const ColumnsEdit = function( { attributes, setAttributes, className, childColumns, updateChildColumnsAlignment } ) {
+	const { columns, verticalAlignment } = attributes;
+
+	// Do any of the child column have a valign setting that is different to what is currently
+	// set on this parent Columns Block?
+	const childColumnHasOveride = !! childColumns.find( ( childColumn ) => childColumn.attributes.verticalAlignment !== verticalAlignment );
+
+	// If we have an overide then remove the setting on the parent so that the toolbar
+	// no longer displays an alignment. This is inline with behaviour of user land editing
+	// tools such as MS Word...etc
+	if ( childColumnHasOveride ) {
+		setAttributes( { verticalAlignment: null } );
```

Comment:
I think this is the reason why `undo` might not work as expected. This should never be part of the render method as it will trigger another re-render when this happens. It also seems like `updateChildColumnsAlignment` will always set `verticalAlignment` to a value, so this will cause immediate re-render given that this condition will be met.

Reply:
Thank you for flagging this. 

The behaviour we want is

* Child Columns "receive" (inherit) value from Parent 
* If Child Column changes its alignment then 
  * that Column uses the alignment that has just been set
  * all _other_ sibling Columns remain "as is" 
  * Parent has its value reset to null (disables in the UI)

This is inline with how most editors handle vertical alignment. How does this marry up with what you're suggesting here https://github.com/WordPress/gutenberg/pull/13899#discussion_r263342659?

Reply:
It fully aligns (no pun intended) with what I shared, I hope :)

> Child Columns "receive" (inherit) value from Parent

That's why when we change something in Parent, we would have to reset all Child Columns instead.

> If Child Column changes its alignment then
>   - that Column uses the alignment that has just been set
>   - all other sibling Columns remain "as is"
>   - Parent has its value reset to null (disables in the UI)

When changing Child Column, the following would have to happen to make that work with the aforementioned behavior:
- all other sibiling Columns get the Parent alignement explicitly set (if set at all)
- the Columns gets updated 
- Parent gets its alignemet unset

Does it make sense? I'm still learning how this might work. Trying to reverse what happens at the moment to avoid this mutation inside render method.

Reply:
Ok I've refactored this so that the child now updates the Parent thereby avoiding the need to mutate during the Parent's render. I should have done this in the first place (end of day coding fai!) so thank you for flagging.

Unfortunately this hasn't solved the undo issue. See screencaptures below which show that the number of undo steps is directly related to the number of Columns. 

#### 2 Cols
![screen capture on 2019-03-07 at 12-04-48](https://user-images.githubusercontent.com/444434/53956019-6ac33a80-40d2-11e9-8b90-209e0177a6c8.gif)

#### 4 Cols
![screen capture on 2019-03-07 at 12-04-09](https://user-images.githubusercontent.com/444434/53956024-6bf46780-40d2-11e9-8617-37b7ff01dc50.gif)

I believe this is because we're updating the attributes of all the child Columns plus the Parent. So it's always `n + 1` steps per v-align update (where `n` is the number of columns). Is there a way to update an attribute without tracking it into history? What other options might we have?

Reply:
@aduth, @youknowriad or @jorgefilipecosta should know more. I'm not aware of any workarounds. For me,  it's the first time where I see where changes need to be applied for both parent and child. We have to solve it as this is something that would be useful for other attributes.

Reply:
A very complex situation. @aduth was working on some improvements to the undo mechanism, maybe the improvements allow some actions to be ignored by undo?
Otherwise, I see some options for this:
Create a new action that allows us to update attributes of multiple blocks, e.g: parent and child, all the siblings etc...
Or we could use the ON_REPLACE action to replace the blocks being updated with themselfs + the attributes updated. This is very "hacky" and I would prefer the other options.

Reply:
I've discussed this with @jorgefilipecosta. We both agree the change is outside the scope of this PR. As a result I will discuss with @aduth and raise and issue if required.

Reply:
Sounds like a good plan 👍